### ISSUE TITLE
🐛 Bug Report: Draw.io Diagram Not Rendering in Backstage TechDocs

### ISSUE THREAD

**Issue by subhanshu-shukla-ril** [backstage/backstage#28007]
### 📜 Description

I am unable to render a Draw.io diagram inside Backstage TechDocs. The diagram renders correctly when using npx @techdocs/cli serve, but fails to display when using yarn dev in the Backstage instance. I have set up the correct MkDocs Docker image with all necessary plugins, but the diagram doesn't show up in Backstage.

### 👍 Expected behavior

The .drawio diagram should render in the Backstage TechDocs instance just like it does when running npx @techdocs/cli serve.

### 👎 Actual Behavior with Screenshots

The .drawio diagram does not render in the Backstage instance when I use yarn dev to run Backstage. The diagram works correctly in npx @techdocs/cli serve, but fails in the Backstage environment.

### 👟 Reproduction steps

I am using the mkdocs-drawio plugin to render Draw.io diagrams, and I have included the external viewer JS (https://viewer.diagrams.net/js/viewer-static.min.js) in the configuration.
It works locally when serving with npx @techdocs/cli serve, but fails to render in the Backstage TechDocs.

Dockerfile for TechDocs Image: I have created a custom Docker image with the following Dockerfile:

# Bug Report: Draw.io Diagram Not Rendering in Backstage TechDocs

## Summary:
I am unable to render a Draw.io diagram inside Backstage TechDocs. The diagram renders correctly when using `npx @techdocs/cli serve`, but fails to display when using `yarn dev` in the Backstage instance. I have set up the correct MkDocs Docker image with all necessary plugins, but the diagram doesn't show up in Backstage.

## Steps to Reproduce:

1. **Dockerfile for TechDocs Image**:
   I have created a custom Docker image with the following Dockerfile:

   ```dockerfile
   FROM python:3.10-alpine

   RUN apk update && apk --no-cache add gcc musl-dev openjdk11-jdk curl graphviz ttf-dejavu fontconfig
   RUN apk --no-cache add git

   # Download plantuml file, validate checksum, and move to the correct location
   RUN curl -o plantuml.jar -L https://github.com/plantuml/plantuml/releases/download/v1.2024.6/plantuml-1.2024.6.jar && echo "3e944755cbed59e1ed9332691d92294bef7bbcda  plantuml.jar" | sha1sum -c - && mv plantuml.jar /opt/plantuml.jar

   RUN pip install --upgrade pip && pip install mkdocs-techdocs-core
   RUN pip install mkdocs-macros-plugin
   RUN pip3 install mkdocs-build-plantuml-plugin
   RUN pip install mkdocs-inline-select-svg-plugin
   RUN pip install mkdocs-drawio

   ENV diagramDir=docs/diagrams

   RUN echo $'#!/bin/sh\n\njava -Dplantuml.include.path=${diagramDir} -jar /opt/plantuml.jar ${@}' > /usr/local/bin/plantuml
   RUN chmod 755 /usr/local/bin/plantuml

   ENTRYPOINT [ "mkdocs" ]


## 2 MkDocs Configuration (mkdocs.yml): 

My mkdocs.yml is configured with the required plugins, including the mkdocs-drawio plugin and the external viewer JS:

```site_name: draw-io-doc
site_description: An informative description
plugins:
  - macros
  - inline-select-svg
  - techdocs-core
  - drawio:
      viewer_js: 'https://viewer.diagrams.net/js/viewer-static.min.js'
nav:
  - Getting Started: index.md
extra:
  unit_price: 10
`

index.md

```
![](catalogue.drawio)

## 3 Running Locally via npx @techdocs/cli serve: 

When I run below command
```npx @techdocs/cli serve --docker-image mkdocs:drawio 
```
The diagram renders perfectly at http://localhost:8000

## Backstage Configuration: 
In the Backstage instance, I have created a Component entity for my documentation as follows:

catalogue.yaml

```
apiVersion: backstage.io/v1alpha1
kind: Component
metadata:
  name: draw-io-doc
  title: draw-io-doc
  annotations:
    backstage.io/techdocs-ref: dir:./draw-io-doc
spec:
  type: documentation
  lifecycle: pre-production
  owner: architects
```

When I run below command in backstage home
```yarn dev
```

Open below url
```
http://localhost:3000/docs/default/component/draw-io-doc
```
Resulting draw-io diagram disappearing 

### 📃 Provide the context for the Bug.

Showing drawio diagram in techdocs

### 🖥️ Your Environment

mac os

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by subhanshu-shukla-ril** [comment]
Can we have someone who can look into it

**Comment by benjdlambert** [comment]
cc @backstage/techdocs-maintainers

**Comment by alexlorenzi** [comment]
Hi @subhanshu-shukla-ril -
I've been looking into this. I couldn't get the draw-io diagram to render fully even using the TechDocs CLI, but I did notice that the mkdocs-plugin seems to be injecting the contents of the drawio file into a `data-mxgraph` attribute in the HTML, can you confirm that's present when you're displaying it in Backstage? 

We have a bunch of sanitization that happens, and I just want to make sure it's not getting cleaned away (I think the plugin injects this after the sanitization, but just to make sure)

Can you also confirm that the DrawIO script is in fact downloaded too, just in case Backstage content policies are blocking anything

**Comment by subhanshu-shukla-ril** [comment]
Just to confirm mkdocs-drawio plugin injecting  drawio script and view div into tech docs, and it is working you open generated index.html inside site folder.
But when you open same tech doc inside backstage  , it deletes the <script> tag of viewer-static.min.js.
I tried with inline script it deletes that too. 
Thats why diagram is not rendering.

**Comment by subhanshu-shukla-ril** [comment]
Any update?

**Comment by johnphilip283** [comment]
Hi @subhanshu-shukla-ril - the techdocs-maintainers will be out until the new year for holiday break - we will get back on this at that point. Thanks for your patience!

**Comment by subhanshu-shukla-ril** [comment]
That means if pass
techdocs:
  sanitizer:
    allowedCustomElementTagNameRegExp: '^script',
    allowedCustomElementAttributeNameRegExp: 'src'

It will allow script tag to render.

**Comment by alexlorenzi** [comment]
Yeah, these configurations are passed down to DOMPurify for sanitation. However, how we attach the  ShadowDOM that contains the rendered TechDoc page prevents script execution within the rendered TechDoc page, mitigating potential security risks

**Comment by subhanshu-shukla-ril** [comment]
Hi @alexlorenzi ,
Can you provide me working config?

**Comment by johnphilip283** [comment]
Hi @subhanshu-shukla-ril - did your above config of:

```
techdocs:
   sanitizer:
      allowedCustomElementTagNameRegExp: '^script',
      allowedCustomElementAttributeNameRegExp: 'src'
```

work in getting it to render?

**Comment by subhanshu-shukla-ril** [comment]
I can see script tag for viewer js but backstage disabled the execution of viewer.js
<img width="695" alt="Image" src="https://github.com/user-attachments/assets/e3db92e8-a1fb-4ebf-9118-1aea9a80d58b" />

**Comment by johnphilip283** [comment]
I see - it's a security risk for Backstage to execute 3rd party JavaScript, so this is probably not the direction we'd recommend. I'd recommend that you follow @alexlorenzi's approach to create a new techdocs addon using the react-drawio library under the hood.
 
> A more robust and secure approach would be to create a [TechDocs addon](https://backstage.io/docs/features/techdocs/addons#creating-an-addon) to handle the diagram rendering from the injected XML attribute. The [react-drawio library](https://github.com/marcveens/react-drawio) and the [mermaid plugin](https://github.com/johanneswuerbach/backstage-plugin-techdocs-addon-mermaid/tree/main) could offer some inspiration for this.

We'd be happy to take a look at an implementation that could be added to https://github.com/backstage/backstage/tree/936e005b3c02d7ac2ce33fb89d2780f86955f921/plugins/techdocs-module-addons-contrib.

**Comment by subhanshu-shukla-ril** [comment]
I have created techdocs module, some this backstage is stopping mx graph to render svg
